doc/doxygen/src/Examples.dox

```diff
@@ -255,6 +287,10 @@
  *     # TextFileSpecification
  * )
  *
+ * # Ensure that the manager is capable of the required workflows.
+ * if not (manager.hasCapability(Manager.Capability.kResolution) and
```

Comment:
[Fix] The resolve check needs to move to the conditional with `ResolvesFutureEntitiesTrait.isImbuedTo`, as we only resolve in that case. You could argue that check is redundant as a manager shouldn't imbue that trait if it doesnt implement resolve, so that check could even be skipped.

Reply:
Yeah this echos my second point in https://github.com/OpenAssetIO/OpenAssetIO/pull/1141#discussion_r1348623413 - when do we eschew `hasCapability` because `managementPolicy` is enough?

The reasoning in this case is that the `managementPolicy` check can be avoided entirely if the capability is not even a possibility.

Reply:
Yeah, but if you do that, youre aborting publishing entirely - which isn't the intended approach, The example is meant to show you can still publish to the path you have. So we can't do the check for resolve here with an `and` with publishing. This  has to illustrate how to work with the variety of managers - some can give you the path up front, some cant.

It doesn't hurt to call `hasCapability before `ResolvesFutureEntitiesTrait.isImbuedTo(policy)` I guess, though it is redundant.  I guess it guards you against a manager messing up its implementation.

Reply:
What do you think of? https://github.com/OpenAssetIO/OpenAssetIO/pull/1141/commits/1dc973e36afe14f4900fda1c04252db9c8354273

We still show capability being a less fine-comb up front precondition for `publishing`, but leave a note saying "hey, see how this trait means you don't need to do a redundant check."

Reply:
Works for me 👍